src/transformers/generation/utils.py

```diff
@@ -1674,6 +1680,9 @@ def generate(
                 logits_processor=prepared_logits_processor,
                 stopping_criteria=prepared_stopping_criteria,
                 pad_token_id=generation_config.pad_token_id,
+                eos_token_id=generation_config.eos_token_id,
```

Comment:
On most decoding methods `eos_token_ids` doesn't need to be passed -- it was used when the decoding method was called directly and `stopping_criteria` was not passed.

However, beam methods still need it.